src/Electrical/Analog/sources.jl

```diff
@@ -1,744 +1,33 @@
-    ]
-    
-    extend(ODESystem(eqs, t, [], pars; name=name), oneport)
-end
-
-"""
-```julia
-function RampCurrent(; name, offset=0.0, start_time=0.0, duration=1.0, height=1.0)
-```
-
-A source in which the current grows linearly from `offset` to `offset+height` over
-the time interval `duration` starting at `start_time`
-
-# States
-- `i(t)`: [`A`]
-  The current through this source
-
-# Connectors
-- `p`
-  Positive port
-- `n`
-  Negative port
-
-# Parameters
-- `offset`: [`A`]
-  A constant offset added to the current output
-- `start_time`: [`s`]
-  The time at which the current starts growing
-- `duration`: [`s`]
-  The duration of the ramp (`0.0` gives a step)
-- `height`: [`A`]
-  The amount that the current grows in the time interval
-"""
-function RampCurrent(;name, offset=0.0, start_time=0.0, duration=1.0, height=1.0)
-    δ = 0.00001
-    @named oneport = OnePort()
-    @unpack v, i = oneport
-    pars = @parameters begin
-        offset=offset
-        height=height 
-        start_time=start_time 
-        duration=duration
-    end
-    eqs = [
-        i ~ _ramp(t, δ, start_time, start_time + duration, height) + offset
-    ]
-    
-    extend(ODESystem(eqs, t, [], pars; name=name), oneport)
-end
-
 """
-```julia
-function SineCurrent(; name, offset=0.0, amplitude=1.0, frequency=1.0, start_time=0.0, phase=0.0)
-```
-
-A source in which the current through its terminals is a sine function of time.
-
+    Voltage(;name)   
 
+Acts as voltage signal.
 
-# States
-- `i(t)`: [`A`]
-  The current through this source
-
-# Connectors
-- `p`
-  Positive port
-- `n`
-  Negative port
-
-# Parameters
-- `offset`: [`A`]
-  A constant offset added to the current output
-- `amplitude`: [`V`]
-  The amplitude of the sine function
-- `frequency`: [`Hz`]
-  The frequency of the sine function
-- `start_time`: [`s`]
-  The time at which the source starts functioning. Before this time, the current through
-  its terminals is `offset`.
-- `phase`: [`rad`]
-  The phase offset of the sine function
 """
-function SineCurrent(;name, offset=0.0, amplitude=1.0, frequency=1.0, start_time=0.0, phase=0.0)
-    δ = 0.00001
-
+function Voltage(;name)   
     @named oneport = OnePort()
     @unpack v, i = oneport
-    pars = @parameters begin
-        offset=offset
-        amplitude=amplitude 
-        frequency=frequency 
-        start_time=start_time 
-        phase=phase
-    end
+    @named V = RealInput()
     eqs = [
-        i ~ _sin_wave(t, frequency, amplitude, start_time, phase) * _step(t, δ, 1.0, start_time) + offset
+        v ~ V.u
     ]
     
-    extend(ODESystem(eqs, t, [], pars; name=name), oneport)
+    extend(ODESystem(eqs, t, [], []; name=name, systems=[V]), oneport)
 end
 
 """
-```julia
-function SquareCurrent(; name, offset=0.0, amplitude=1.0, frequency=1.0, start_time=0.0)
-```
+    Current(;name)   
 
-A source in which the current through its terminals is a square function of time.
+Acts as current signal.
```

Comment:
Add a description about the convention of a positive current, i.e. which port has inflowing current if `I` is positive.

Reply:
Also here, it is an ideal current source with no internal resistance.